source/input.c

```diff
@@ -0,0 +1,68 @@
+#ifndef STRUCTS
+#define STRUCTS
+
+#include "../headers/structs.h"
+
+#endif
+
+#ifndef FINAL
```

Comment:
```suggestion
#ifndef INPUT
```

Reply:
The define constant must show, which header file it corresponds with, so it is easier to find furter on